src/modules/Game.class.js

```diff
@@ -20,25 +20,78 @@
    * If passed, the board will be initialized with the provided
    * initial state.
    */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.state = 'idle';
+    this.score = 0;
+    this.initialState = initialState;
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
+  // eslint-disable-next-line no-console
 
+  moveLeft() {
+    if (this.state === 'idle') {
+      return;
+    }
+
+    const updatedState = this.processMoveLeft();
+
+    this.updatedGameState(updatedState);
+
+    return this.initialState;
+  }
+  moveRight() {
+    if (this.state === 'idle') {
+      return;
+    }
+
+    const updatedState = this.processMoveRight();
+
+    this.updatedGameState(updatedState);
+
+    return this.initialState;
+  }
+  moveUp() {
+    if (this.state === 'idle') {
+      return;
+    }
+
+    const updatedState = this.processMoveUp();
+
+    this.updatedGameState(updatedState);
+
+    return this.initialState;
+  }
+  moveDown() {
+    if (this.state === 'idle') {
```

Comment:
The condition should be `this.state === 'playing'` to ensure moves are only allowed when the game is actively being played.